src/ToDo.UI/Views/WelcomePage.xaml

```diff
@@ -1,28 +1,116 @@
-	xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
-	xmlns:uen="using:Uno.Extensions.Navigation.UI"
-	xmlns:uer="using:Uno.Extensions.Reactive.UI"
-	xmlns:utu="using:Uno.Toolkit.UI"
-	xmlns:um="using:Uno.Material"
-	mc:Ignorable="d"
-	Background="{ThemeResource MaterialBackgroundBrush}">
+﻿<Page x:Class="ToDo.Views.WelcomePage"
+	  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+	  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+	  xmlns:local="using:ToDo.Views"
+	  xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	  xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	  xmlns:uen="using:Uno.Extensions.Navigation.UI"
+	  xmlns:uer="using:Uno.Extensions.Reactive.UI"
+	  xmlns:utu="using:Uno.Toolkit.UI"
+	  xmlns:um="using:Uno.Material"
+	  mc:Ignorable="d"
+	  Background="{ThemeResource MaterialBackgroundBrush}">
 
-	<utu:AutoLayout Padding="10,10,10,10">
-		<utu:AutoLayout Padding="20,60,20,60" Orientation="Horizontal" utu:AutoLayout.PrimaryAlignment="Stretch">
-			<Image Source="ms-appx:///Assets/Welcome/welcome_uno.png" Stretch="Uniform" utu:AutoLayout.CounterAlignment="Start" utu:AutoLayout.PrimaryAlignment="Stretch" utu:AutoLayout.CounterLength="310" />
+	<Grid>
+		<VisualStateManager.VisualStateGroups>
+			<VisualStateGroup>
+				<VisualState x:Name="Wide">
+					<VisualState.StateTriggers>
+						<AdaptiveTrigger MinWindowWidth="{StaticResource WideMinWindowWidthh}" />
+					</VisualState.StateTriggers>
+					<VisualState.Setters>
+						<Setter Target="NarrowWelcomeContent.Visibility"
+								Value="Collapsed" />
+						<Setter Target="WideWelcomeContent.Visibility"
+								Value="Visible" />
+					</VisualState.Setters>
+				</VisualState>
+
+				<VisualState x:Name="Narrow">
+					<VisualState.StateTriggers>
+						<AdaptiveTrigger MinWindowWidth="0" />
+					</VisualState.StateTriggers>
+					<VisualState.Setters>
+						<Setter Target="NarrowWelcomeContent.Visibility"
+								Value="Visible" />
+						<Setter Target="WideWelcomeContent.Visibility"
+								Value="Collapsed" />
+					</VisualState.Setters>
+				</VisualState>
+			</VisualStateGroup>
+		</VisualStateManager.VisualStateGroups>
+
+		<utu:AutoLayout x:Name="NarrowWelcomeContent"
+						Visibility="Collapsed"
+						Padding="10,10,10,10">
+			<utu:AutoLayout Padding="20,60,20,60"
+							Orientation="Horizontal"
+							utu:AutoLayout.PrimaryAlignment="Stretch">
+				<Image Source="ms-appx:///Assets/Welcome/welcome_uno.png"
+					   Stretch="Uniform"
+					   utu:AutoLayout.CounterAlignment="Start"
+					   utu:AutoLayout.PrimaryAlignment="Stretch"
+					   utu:AutoLayout.CounterLength="310" />
+			</utu:AutoLayout>
+			<utu:AutoLayout Spacing="20"
+							Padding="0,40,0,40">
+				<TextBlock Foreground="{ThemeResource MaterialOnSecondaryMediumBrush}"
+						   TextAlignment="Center"
+						   TextWrapping="Wrap"
+						   Text="Welcome to&#xA;Uno To Do!"
```

Comment:
not sure if it is a good idea to keep newline literal in them, with text wrap
shouldnt we just pick one not both?

Reply:
@carldebilly what do you think ? regarding the figma plugin part for the generated code